### PR TITLE
Upgrade NPM packages to support NodeJS 12

### PR DESCRIPTION
**Pull request by cocreature**

### CODE REVIEW

File: test/validation/AllObjects.spec.ts [hunk 1]
```diff
@@ -165,16 +165,20 @@ function test<A extends { [_: string]: any }>(
                 arbitrary,
                 value => {
                     const key = pickFrom(keys);
-                    value[key] = null;
-                    return !isValid(validation.validate(value));
+                    // upcast to any so we can produce an invalid value.
```
Comment (cocreature) [inline test/validation/AllObjects.spec.ts:168]: TypeScript 3.5 added stricter checks on assignments which broke this. Given that we are deliberately doing invalid stuff here, casting to `any` seemed like a reasonable option but I’m happy to learn others.

File: package.json [hunk 2]
```diff
@@ -44,29 +44,29 @@
...
-    "protobufjs": "6.8.8",
-    "sinon": "7.3.1",
-    "ts-node": "8.0.3",
-    "typedoc": "0.15.0",
-    "typescript": "3.4.3",
-    "uuid": "3.3.2"
+    "@types/chai": "^4.2.12",
+    "@types/google-protobuf": "^3.7.2",
+    "@types/mocha": "^8.0.1",
+    "@types/node": "~12.0.12",
```
[the remark this answers is not in the document]
Comment (stefanobaghino-da) [reply package.json:57]: Cool, but next time please use exact dependency versions.

[the remark this answers is not in the document]
Comment (cocreature) [reply package.json:57]: Why? I thought that’s what you have the lockfile for.

File: test/validation/AllObjects.spec.ts [hunk 3]
```diff
@@ -188,16 +192,20 @@ function test<A extends { [_: string]: any }>(
...
         });
         jsc.property(
             'signal an unexpected key error on objects with an extra key',
             arbitrary,
             value => {
                 const extraKey = 'supercalifragilisticexpialidocious'; // reasonably no one will ever use this as a key
-                value[extraKey] = null;
-                return containsError(validation.validate(value).errors, {
+                // upcast to any so we can produce an invalid value.
+                const genericValue: any = value;
```
Comment (stefanobaghino-da) [inline test/validation/AllObjects.spec.ts:206]: I would usually recommend `unknown` for this kind of things. Not very important though.

**Pull request merged**